src/client.rs

```diff
@@ -29,6 +29,12 @@ impl Plugin for ClientPlugin {
                 PreUpdate,
                 ClientSet::Receive.after(NetcodeClientPlugin::update_system),
             )
+            .configure_sets(
+                PreUpdate,
+                ClientSet::Reset
+                    .after(NetcodeClientPlugin::update_system)
+                    .run_if(bevy_renet::client_just_disconnected()),
```

Comment:
Didn't know that we have this condition. Probably worth checking other places (not necessary as part of this PR).